src/scenes/mainMenu.js

```diff
@@ -1,40 +1,58 @@
-const btnImageSpritesheet = "assets/textures/mainMenu/buttons.png";
-const btnScenes = [
-	"Challenge Mode",
-	"Practice Mode",
-	"Instructions",
-	"Cheat Sheet",
-	"Settings"
-];
-let btnImgs = [];
-let btns = [];
+import Scene from "../objects/scene.js";
+import { scaleNearestNeighbor } from "../helperFunctions/imageFunctions.js";
+import ImageButton from "../objects/imageButton.js";
+import sceneManager from "../objects/sceneManager.js";
+import canvasManager from "../objects/canvasManager.js";
+import { CHALLENGE_MODE, PRACTICE_MODE, INSTRUCTIONS, CHEAT_SHEET, SETTINGS } from "../constants/sceneConstants.js";
 
-const mainMenu = {
-    "setup": function()
-    {
-    	btnImgs = [];
-    	btns = [];
-    	let img = loadImage(btnImageSpritesheet, () => {
-    		img.loadPixels();
-    		btnScenes.forEach((e, i, a) => {
-				let subImg = img.get(0, i*11, 75, 11);
-    			btnImgs.push(scaleNearestNeighbor(subImg, 480, 70));
-    		});
-    		btnImgs.forEach((e, i, a) => {
-    			let btn = new ImageButton(e, 0, i*70, () => {
-    				currentScene = btnScenes[i];
-    				btns.forEach((e, i, a) => {
-    					e.destroy();
-    				});
-    				scenes[currentScene]["setup"]();
-    			});
-    			btn.init();
-    			btns.push(btn);
-    		});
-    	});
-    },
-    "draw": function()
-    {
-        background(255, 0, 255);
+export default class MainMenu extends Scene {
+    constructor() { 
+        super(); 
+        this.imageButtons = [];
     }
-}
+
+	/**
+	 * Our own private method for changing scenes.
+	 * @param {string} sceneName The name of the scene we want to change to.
+	 */
+	#changeScene(sceneName) {
+		this.imageButtons = [];
+		sceneManager.setCurrentScene(sceneName);
+	}
+
+    setup() {
```

Comment:
I feel like we might be able to improve the implementation here as well, but that was out of scope for this ticket.